OpenRA.Mods.Common/Activities/Enter.cs

```diff
@@ -36,15 +37,28 @@ enum EnterState { ApproachingOrEntering, Inside, Exiting, Done }
 		Activity inner;
 		bool firstApproach = true;
 
-		protected Enter(Actor self, Actor target, EnterBehaviour enterBehaviour, int maxTries = 1, bool repathWhileMoving = true)
+		/// <summary>
+		/// Make an actor enter the target actor.
+		/// </summary>
+		/// <param name="enterRange">
+		///		When the distance between the target actor and self is closer than or equal to this range,
+		///		self will enter the target. Non-zero range allows self to enter a moving target.
+		/// </param>
+		protected Enter(Actor self, Actor target, EnterBehaviour enterBehaviour, WDist enterRange,
+			int maxTries = 1, bool repathWhileMoving = true)
 		{
 			move = self.Trait<IMove>();
 			this.target = Target.FromActor(target);
 			this.maxTries = maxTries;
 			this.enterBehaviour = enterBehaviour;
 			this.repathWhileMoving = repathWhileMoving;
+			this.enterRange = enterRange;
 		}
 
+		protected Enter(Actor self, Actor target, EnterBehaviour enterBehaviour,
+			int maxTries = 1, bool repathWhileMoving = true) :
+			this(self, target, enterBehaviour, WDist.Zero, maxTries, repathWhileMoving) {}
```

Comment:
There needs to be a space between the {} brackets:

>OpenRA.Mods.Common/Activities/Enter.cs:L60: [SA1012] Invalid spacing around the opening curly bracket.
OpenRA.Mods.Common/Activities/Enter.cs:L60: [SA1013] Invalid spacing around the closing curly bracket.